app/src/main/java/com/android/sample/ui/account/AccountScreen.kt

```diff
@@ -158,7 +159,11 @@ private fun ListSelection(
           navigationActions.navigateTo(Screen.OVERVIEW_RECIPE_ACCOUNT)
         },
         topCornerButton = { recipe ->
-          if (selectedListIndex == 0) TopCornerLikeButton(recipe, userViewModel)
+          if (selectedListIndex == 0) {
+            TopCornerLikeButton(recipe, userViewModel, true)
+          } else {
+            TopCornerLikeButton(recipe, userViewModel, false)
+          }
```

Comment:
This is a continuation of my previous comment. You could simplify and improve readability by splitting the functionality into two separate methods of this file: `app/src/main/java/com/android/sample/ui/utils/RecipeList.kt`. 
Here's a suggestion:
```suggestion
          if (selectedListIndex == 0) {
            TopCornerDeleteButton(recipe, userViewModel)
          } else {
            TopCornerLikeButton(recipe, userViewModel)
          }
```

Reply:
Done